src-built-in/components/workspaceManagementMenu/src/stores/workspaceManagementMenuStore.js

```diff
@@ -326,19 +332,21 @@ Actions = {
 	 * Asks the user if they'd like to save their data, then loads the requested workspace.
 	 */
 	switchToWorkspace: function (data) {
-		if (switching) return;
-		switching = true;
+		if (Actions.getIsSwitchingWorkspaces()) return;
+		Actions.setIsSwitchingWorkspaces(true);
```

Comment:
@d4hines you're changing from a simple boolean to using a store. That fundamentally complicates the logic, opening up the possibility of introducing race conditions. This task didn't require changing this code.